analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextBaseTest.ShouldAnalyzeTree.cs

```diff
@@ -71,6 +71,21 @@ public void ShouldAnalyzeTree_Scanner_UnchangedFiles_ContainsOtherFile()
         ShouldAnalyzeTree(options).Should().BeTrue();
     }
 
+    [TestMethod]
+    public void ShouldAnalyzeTree_Scanner_WhenRazorFileHasNotChanged_ReturnsFalseForTheAssociatedGeneratedFile()
+    {
+        // The generated files have a different root that is not absolute and might not exist on disk.
+        const string generatedFileName = "Microsoft.NET.Sdk.Razor.SourceGenerators\\Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator\\Pages_Component_razor.g";
+        const string content =
+            """
```

Comment:
Nitpick: from the rest of the code base, it seems we usually put the opening `"""` next to the `=` equal sign.
```
 const string content = """
            #pragma ...
            """;
```

Reply:
I find it easier to read in this way. I wouldn't change it.